modules/storage-page-memory/src/main/java/org/apache/ignite/internal/storage/pagememory/mv/WriteIntentListSupport.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal.storage.pagememory.mv;
+
+import static org.apache.ignite.internal.pagememory.util.PageIdUtils.NULL_LINK;
+
+import java.util.function.Supplier;
+import org.apache.ignite.internal.lang.IgniteInternalCheckedException;
+import org.apache.ignite.internal.pagememory.freelist.FreeListImpl;
+import org.apache.ignite.internal.storage.StorageException;
+
+class WriteIntentListSupport {
+    static void removeNodeFromWriteIntentsList(
```

Comment:
First of all, some "pictures" would definitely help for these algorithms.

Second, I see that you update previous page and next page, but you don't update "current" page, which makes this all operation inconsistent. Could you please explain what I'm missing?

Reply:
I fixed the inconsistency. Also, I added a javadoc with pictures (stolen from the ticket) to `WiLinkableRowVersion`